pulsar_broker_tests are consistently failing on an arm64 mac

**Describe the bug**
Every time I run the tests on my M1-base mac pulsar_broker_tests are failing with the following message:

```
failures:

---- source::pulsar_source::pulsar_broker_tests::test_doc_batching_logic stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_doc_batching_logic' panicked at 'Setup pulsar source: Connection error: fatal error when connecting to the Pulsar server

Caused by:
    fatal error when connecting to the Pulsar server', quickwit-indexing/src/source/pulsar_source.rs:834:14

---- source::pulsar_source::pulsar_broker_tests::test_multi_topic_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_multi_topic_ingestion' panicked at 'Create source: Failed to create source `test-pulsar-source--source-RPFBm` of type `pulsar`. Cause: Connection error: fatal error when connecting to the Pulsar server

Caused by:
    fatal error when connecting to the Pulsar server

Caused by:
    0: Connection error: fatal error when connecting to the Pulsar server
    1: fatal error when connecting to the Pulsar server', quickwit-indexing/src/source/pulsar_source.rs:968:10

---- source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_multi_consumer_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_multi_consumer_ingestion' panicked at 'Send admin request: reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8081), path: "/admin/v2/persistent/public/default/test-pulsar-source--partitioned-multi-consumer--topic-gTKk2/partitions", query: None, fragment: None }, source: hyper::Error(IncompleteMessage) }', quickwit-indexing/src/source/pulsar_source.rs:700:14

---- source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_single_consumer_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_single_consumer_ingestion' panicked at 'Send admin request: reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8081), path: "/admin/v2/persistent/public/default/test-pulsar-source--partitioned-single-consumer--topic-HdfgV/partitions", query: None, fragment: None }, source: hyper::Error(IncompleteMessage) }', quickwit-indexing/src/source/pulsar_source.rs:700:14

---- source::pulsar_source::pulsar_broker_tests::test_topic_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_topic_ingestion' panicked at 'Create source: Failed to create source `test-pulsar-source--source-GQr9o` of type `pulsar`. Cause: Connection error: fatal error when connecting to the Pulsar server

Caused by:
    fatal error when connecting to the Pulsar server

Caused by:
    0: Connection error: fatal error when connecting to the Pulsar server
    1: fatal error when connecting to the Pulsar server', quickwit-indexing/src/source/pulsar_source.rs:911:10


failures:
    source::pulsar_source::pulsar_broker_tests::test_doc_batching_logic
    source::pulsar_source::pulsar_broker_tests::test_multi_topic_ingestion
    source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_multi_consumer_ingestion
    source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_single_consumer_ingestion
    source::pulsar_source::pulsar_broker_tests::test_topic_ingestion

test result: FAILED. 130 passed; 5 failed; 6 ignored; 0 measured; 0 filtered out; finished in 8.70s

error: test failed, to rerun pass `-p quickwit-indexing --lib`
make[1]: *** [test-all] Error 101
make: *** [test-all] Error 2
```

**Steps to reproduce (if applicable)**
Steps to reproduce the behavior:

1. Run make test-all on M1 mac
2. Observe the error message above

**Expected behavior**
Tests should pass

**Configuration:**
N/A

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -74,8 +74,8 @@ services:
       retries: 100
 
   pulsar-broker:
-    # The oldest supported version. EOL January 10, 2024
-    image: apachepulsar/pulsar:${PULSAR_VERSION:-2.11.1}
+    # The oldest version with arm64 docker images. EOL May 2 2025
+    image: apachepulsar/pulsar:${PULSAR_VERSION:-3.0.0}
     container_name: pulsar-broker
     command: bin/pulsar standalone
     ports:

